Refactor 0.7 Deployment And Login Page

argha-dot opened the pull request:
Created a shell deployment and login page, in leptos 0.7

argha-dot committed:
hello world.....

argha-dot committed:
added input component

argha-dot committed:
fix scss issue

argha-dot committed:
init login page, created link, button, page_container, input, password input components

argha-dot committed:
login page with minor form validation

argha-dot committed:
deployment dashboard shell

rakshith-ravi committed:
Removed warnings

rakshith-ravi commented on .cargo/config.toml:
```diff
@@ -2,10 +2,15 @@
 
 [target.'cfg(target_arch = "wasm32")']
-rustflags = ["--cfg=getrandom_backend=\"wasm_js\""]
+rustflags = ["--cfg", "getrandom_backend=\"wasm_js\""]
```
We don't need both. Remove one of them

rakshith-ravi commented on .devcontainer/devcontainer.json:
Retain tabs for this file please

argha-dot replied:
no

rakshith-ravi commented on config/tailwind.config.js:
I thought we don't need this file anymore?

rakshith-ravi commented on frontend/Cargo.toml:
Do confirm if we still need `leptos_meta`. I remember reading somewhere that because of the `shell` thing in 0.7, we might not need it anymore

argha-dot replied:
It's there in hackernews_axum in leptos axum

rakshith-ravi commented on frontend/src/app.rs:
```diff
@@ -1,141 +1,11 @@
-				/>
-			</Routes>
-		</Router>
+		<TempPageContainer>
```
I wonder if it makes sense to keep a single one and use a `children` prop for the outlet

argha-dot replied:
no, anyway we're gonna use a router, then I can just remove this, instead of make edits to the component

rakshith-ravi commented on frontend/src/components/container/main_container/container_body.rs:
Why are these new files instead of changed files?

argha-dot replied:
I made these changes in different commits, so maybe that's why

rakshith-ravi commented on frontend/src/components/link.rs:
```diff
@@ -1,88 +1,40 @@
+	variant: Signal<LinkStyleVariant>,
+	/// The Target of the Link, to be used with the link variant
+	#[prop(into, optional)]
+	to: Signal<String>,
```
Leave a TODO for this:

Ideally, this would be a struct that points to the next location's URL, (through the `declare_app_route` macro)

argha-dot replied:
what if we want to navigate to an external route? say a github link

argha-dot replied:
opened #76

rakshith-ravi commented on frontend/src/components/mod.rs:
```diff
@@ -1,134 +1,21 @@
-/// The tooltip component is used to display a tooltip when the user hovers over
-/// an element. It is used to show the user more information about an element.
-pub mod tooltip;
+mod alert;
+mod button;
+mod container;
+mod icon;
+mod input;
+mod link;
+mod page_title;
+mod password_input;
+mod status_badge;
```
Add documentation for these. Copy them from the old one if you want, and use Copilot to improve them

argha-dot replied:
most of these are files which have a documented single component that's being exported, so naming them doesn't make any sense, especially since all of them are being exposed via prelude, and not accessed directly.

rakshith-ravi commented on frontend/src/lib.rs:
```diff
@@ -92,45 +43,48 @@ pub fn hydrate() {
+				<AutoReload options=options.clone() />
+				<HydrationScripts options/>
+
+				<title>"Patr"</title>
```
Let's make this `leptos_meta::Title`

rakshith-ravi commented on frontend/src/pages/deployment/dashboard/components/mod.rs:
```diff
@@ -0,0 +1,78 @@
+#[derive(Clone)]
+pub struct DeploymentCardItem {
+	/// The Label of the deployment
+	label: &'static str,
```
I'm guessing these are gonna change?

argha-dot replied:
yeah, the value field will change to a String

argha-dot replied:
I'll make this change now

rakshith-ravi commented on frontend/src/utils/color.rs:
```diff
@@ -82,3 +82,28 @@ impl TextColor {
+/// All Tailwind variants that take colors
+#[derive(Debug, Clone, Copy, PartialEq, Eq, strum::Display)]
+#[strum(serialize_all = "kebab-case")]
+pub enum TailwindClass {
```
Ignoring this for now, but I feel like the enum name would change soon

argha-dot replied:
yeah, I had an idea, I'm not able to make it sense now, I'll finish this later

argha-dot committed:
fix PR changes
 - remove tailwind config file
 - changes indent for devcontainer.json
 - use leptos_meta for title
 - other misc changes

argha-dot committed:
added documentation for all modules

argha-dot committed:
fixed variant renaming

argha-dot committed:
fix icks
